eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -0,0 +1,240 @@
+            annotationMetadata = new AnnotationMetadata(annotationVepVersion, annotationVepCacheVersion);
+        }
+
+        if (alternate != null) {
+            return service.findByChromosomeAndStartAndReferenceAndAlternate(regionId[0], Integer.parseInt(regionId[1]),
+                    regionId[2], alternate, annotationMetadata);
+        } else {
+            return service.findByChromosomeAndStartAndReference(regionId[0], Integer.parseInt(regionId[1]),
+                    regionId[2], annotationMetadata);
+        }
+    }
+
+    @GetMapping(value = "/{variantCoreString}/annotations")
+    public QueryResponse getAnnotations(@PathVariable("variantCoreString") String variantCoreString,
+                                        @RequestParam(name = "species") String species,
+                                        @RequestParam(name = "assembly") String assembly,
+                                        @RequestParam(name = "annot-vep-version", required = false)
+                                                String annotationVepVersion,
+                                        @RequestParam(name = "annot-vep-cache-version", required = false)
+                                                String annotationVepCacheVersion,
+                                        HttpServletResponse response) throws IllegalArgumentException {
+        initializeQuery();
+
+        try {
+            checkParameters(variantCoreString, annotationVepVersion, annotationVepCacheVersion,
+                    species, assembly);
+        } catch (IllegalArgumentException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setErrorQueryResponse(e.getMessage());
+        }
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
+
+        List<VariantWithSamplesAndAnnotation> variantEntities;
+        try {
+            variantEntities = getVariantEntitiesByParams(variantCoreString, annotationVepVersion,
+                    annotationVepCacheVersion);
+            if (variantEntities.size() == 0) {
+                response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                return setQueryResponse(buildQueryResult(variantEntities, 0));
+            }
+        } catch (AnnotationMetadataNotFoundException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setQueryResponse(ex.getMessage());
+        }
+
+        List<Annotation> annotations = new ArrayList<>();
+        variantEntities.forEach(variantEntity -> {
+            annotations.add(variantEntity.getAnnotation());
+        });
+
+        QueryResult<Annotation> queryResult = buildQueryResult(annotations, annotations.size());
+
+        return setQueryResponse(queryResult);
+    }
+
+    @GetMapping(value = "/{variantCoreString}/source-entries")
+    public QueryResponse getsources(@PathVariable("variantCoreString") String variantCoreString,
```

Comment:
Rename this method as `getSources`. Also, if the HATEOAS link is called `sources`, it would be reasonable to make the URL `/{variantCoreString}/sources` too.

Reply:
Fixed :+1: